jvm.c

```diff
@@ -1138,8 +1196,24 @@ stack_entry_t *execute(method_t *method,
 
             char *class_name = find_class_name_from_index(index, clazz);
             class_file_t *target_class;
-            if (find_or_add_class_to_heap(class_name, prefix, &target_class)) {
-                /* Call static initialization */
+
+            /* FIXME: use linked list to prevent wasted space */
+            class_file_t **stack = malloc(sizeof(class_file_t *) * 100);
+            size_t count = 0;
+            do {
```

Comment:
Rewrite as following:
```c
            while (true) {
                ...
                class_name = find_class_name_from_index(
                    target_class->info->super_class, target_class);
                if (!strcmp(class_name, "java/lang/Object"))
                    break;
            }
```

Or equivalent for iteration:
```c
for (char *class_name = find_class_name_from_index(index, clazz); true; ) {
    ...
}
```

It would be more intuitive for understanding.